packages/docs/src/components/theme-toggle/theme-toggle.tsx

```diff
@@ -0,0 +1,73 @@
+import { $, component$, useClientEffect$, useContext, useStyles$ } from '@builder.io/qwik';
+import { SunAndMoon } from './sun-and-moon';
+import { themeStorageKey } from '../router-head/theme-script';
+import themeToggle from './theme-toggle.css';
+import { GlobalStore } from '../../context';
+
+export type ThemePreference = 'dark' | 'light';
+
+export const colorSchemeChangeListener = (onColorSchemeChange: (isDark: boolean) => void) => {
+  const listener = ({ matches: isDark }: MediaQueryListEvent) => {
+    onColorSchemeChange(isDark);
+  };
+  window
+    .matchMedia('(prefers-color-scheme: dark)')
+    .addEventListener('change', (event) => listener(event));
+
+  return () =>
+    window.matchMedia('(prefers-color-scheme: dark)').removeEventListener('change', listener);
+};
+
+export const setPreference = (theme: ThemePreference) => {
+  localStorage.setItem(themeStorageKey, theme);
+  reflectPreference(theme);
+};
+
+export const reflectPreference = (theme: ThemePreference) => {
+  document.firstElementChild?.setAttribute('data-theme', theme);
+};
+
+export const getColorPreference = (): ThemePreference => {
+  if (localStorage.getItem(themeStorageKey)) {
+    return localStorage.getItem(themeStorageKey) as ThemePreference;
+  } else {
+    return window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+  }
+};
+
+export const ThemeToggle = component$(() => {
+  useStyles$(themeToggle);
+  const state = useContext(GlobalStore);
+
+  useClientEffect$(() => {
+    state.theme = getColorPreference();
+  });
+
+  useClientEffect$(() => {
+    return colorSchemeChangeListener((isDark) => {
+      state.theme = isDark ? 'dark' : 'light';
+      setPreference(state.theme);
+    });
+  });
```

Comment:
there is one case when this not work, because of component is not rendered yet. After openning the page on a mobile, changing preference to light/dark mode will not have no effect. It is ok? I think to fix this I can move this code to another "hidden" component always rendered for both - desktop and mobile (or there is any better solution?)